### DISCUSSION TITLE
How to force redeploy with pulling docker image?

### DISCUSSION THREAD

**Discussion by lolhax1337** [argoproj/argo-cd#10099] [Q&A]
Hi,

All I want is to redeploy deployment using argocd CLI. Expected behavior is like action:
`kubectl rollout restart deployment <my_deployment>`

When deployment is restarted and docker image is downloading again. Is it possible in argocd?

**Comment by byt3-m3** [comment]
+1, Id like to know as well? I can only do this now by changing the image tag in my dev enviornment, but if the image is set to the latest in my prod env, I would expect argo to pull the image during a sync.

**Comment by washanhanzi** [comment]
I think the only way to do this is using [`image updater`](https://argocd-image-updater.readthedocs.io/en/stable/) with `digest` update strategy.

**Comment by kostis-codefresh** [comment]
If you are using "latest" for a tag this is a bad practice and you should avoid it (especially in production)

**Comment by Babbili** [comment]
in your deployment template containers spec, if you define `imagePullPolicy: Always` it will always pull it when redeploy

>   imagePullPolicy	<string>
    Image pull policy. One of Always, Never, IfNotPresent. Defaults to Always if
    :latest tag is specified, or IfNotPresent otherwise. Cannot be updated. More
    info: https://kubernetes.io/docs/concepts/containers/images#updating-images
    
    Possible enum values:
     - `"Always"` means that kubelet always attempts to pull the latest image.
    Container will fail If the pull fails.
     - `"IfNotPresent"` means that kubelet pulls if the image isn't present on
    disk. Container will fail if the image isn't present and the pull fails.
     - `"Never"` means that kubelet never pulls an image, but only uses a local
    image. Container will fail if the image isn't present

**Reply by washanhanzi** [reply]
I don't think argo will restart the pod if there is no change to the image tag or app definition. Even if argo re-apply the yaml config, k8s won't restart the pod  if there is nothing changed.
